### PR TITLE
Documentation refactoring

### PR DESCRIPTION
**Pull request by l0k0ms**
### What does this PR do?

This is a first attempt at enhancing and increasing the documentation around the Datadog-operator.

As a first step no new content is created, it's only enhanced and layed out differently to allow an easier consumption.

Other PRs are going to follow to give more details and abstract more the example given.

### Motivation

Prepare the beta

### CODE REVIEW

File: docs/cluster_agent_setup.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+$ kubectl apply -n $DD_NAMESPACE -f datadog-agent-with-clusteragent.yaml
+datadogagent.datadoghq.com/datadog-agent configured
+```
+
+Verify that the Agent and Cluster Agent are correctly running:
+
+```shell
+$ kubectl get -n $DD_NAMESPACE dd datadog-agent
+NAME            ACTIVE   AGENT     CLUSTER-AGENT   AGE
+datadog-agent   True     Running   Running         15m22s
```
Comment (ahmed-mez) [inline docs/cluster_agent_setup.md]: We updated this, `kubectl get -n $DD_NAMESPACE dd datadog-agent` shows current state (Progressing, Running , Updating , Failed) with the values in the tuple representing (Requested / Ready / Uptodate)

```
NAME            ACTIVE   AGENT             CLUSTER-AGENT     CLUSTER-CHECKS-RUNNER   AGE
datadog-agent   True     Running (2/2/2)   Running (2/2/2)                           6m
```

File: docs/custom_check.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+Once the `ConfigMaps` are configured, a `DatadogAgent` resource can be created to use them with the following chart:
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog-agent
+spec:
+  credentials:
+    apiKey: <DATADOG_API_KEY>
```
Comment (ahmed-mez) [inline docs/custom_check.md]: The app key is required to forward metrics and events to datadog

```suggestion
    apiKey: <DATADOG_API_KEY>
    appKey: <DATADOG_APP_KEY>
```

File: docs/custom_check.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+Additional user-configured volumes can be mounted in either the node or Cluster Agent containers by setting the `volumes` and `volumeMounts` properties. Find below an example of using a volume to mount a secret:
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog-agent
+spec:
+  credentials:
+    apiKey: <DATADOG_API_KEY>
```
Comment (ahmed-mez) [inline docs/custom_check.md]:
```suggestion
    apiKey: <DATADOG_API_KEY>
    appKey: <DATADOG_APP_KEY>
```

File: docs/getting_started.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+The following [`datadog-agent.yaml` file](https://github.com/DataDog/datadog-operator/blob/master/examples/datadog-agent.yaml) is the simplest configuration for the Datadog Operator:
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog-agent
+spec:
+  credentials:
+    apiKey: <DATADOG_API_KEY>
```
Comment (ahmed-mez) [inline docs/getting_started.md]:
```suggestion
    apiKey: <DATADOG_API_KEY>
    appKey: <DATADOG_APP_KEY>
```

File: docs/getting_started.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+```shell
+$ kubectl apply -n $DD_NAMESPACE -f datadog-agent.yaml
+datadogagent.datadoghq.com/datadog-agent created
+```
+
+You can check the state of the `DatadogAgent` ressource with:
+
+```shell
+kubectl get -n $DD_NAMESPACE datadogagent datadog-agent
```
Comment (ahmed-mez) [inline docs/getting_started.md]: nit
```suggestion
kubectl get -n $DD_NAMESPACE dd datadog-agent
```

File: docs/getting_started.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+NAME             ACTIVE   AGENT     CLUSTER-AGENT   AGE
+datadog-agent    True     Running                   4m2s
```
Comment (ahmed-mez) [inline docs/getting_started.md]:
```
NAME            ACTIVE   AGENT             CLUSTER-AGENT   CLUSTER-CHECKS-RUNNER   AGE
datadog-agent   True     Running (2/2/2)                                           110m
```

File: docs/getting_started.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+Update your [`datadog-agent.yaml` file](https://github.com/DataDog/datadog-operator/blob/master/examples/datadog-agent-with-tolerations.yaml) with the following configuration to add the toleration in the `Daemonset.spec.template` of your `DaemonSet` :
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog-agent
+spec:
+  credentials:
+    apiKey: <DATADOG_API_KEY>
```
Comment (ahmed-mez) [inline docs/getting_started.md]:
```suggestion
    apiKey: <DATADOG_API_KEY>
    appKey: <DATADOG_APP_KEY>
```

File: README.md [hunk 8]
```diff
@@ -10,70 +10,42 @@
...
-
-* Agent configuration validation that limits configuration mistakes.
-* Orchestration of creating/updating Datadog Agent resources.
-* Reporting of Agent configuration status in its Kubernetes CRD resource.
-* Optionally, use of on an advanced `DaemonSet` deployment by leveraging the [ExtendedDaemonSet](https://github.com/DataDog/extendeddaemonset).
-* Many other features to come :).
+- Agent configuration validation that limits configuration mistakes.
+- Orchestration of creating/updating Datadog Agent resources.
+- Reporting of Agent configuration status in its Kubernetes CRD resource.
+- Optionally, use of on an advanced `DaemonSet` deployment by leveraging the [ExtendedDaemonSet][2].
```
Comment (cswatt) [inline README.md]:
```suggestion
- Optionally, use of an advanced `DaemonSet` deployment by leveraging the [ExtendedDaemonSet][2].
```

File: README.md [hunk 9]
```diff
@@ -10,70 +10,42 @@
...
+The official [Datadog Helm chart][3] is still the recommended way to setup Datadog in a Kubernetes cluster, as it has most supported configuration options readily accessible. It also makes using more advanced features easier than rolling your own deployments.
 
-## Quick start
+The **Datadog Operator** aims to improve the user experience around deploying Datadog. It does this by reporting deployment status, health, and errors in its Custom Resource status, and by limiting the risk of misconfiguration thanks to higher-level configuration options.
 
-See the [Quick Start tutorial](docs/quick-start.md).
+However, the Datadog Operator is still in alpha, so it is not yet a recommended way of installing the Agent in production. Our end goal is to support both the Helm chart and the operator (as well as providing basic deployment files in the Agent [repository][4] when CRDs and Helm are not an option) as official ways of installing Datadog.
```
Comment (cswatt) [inline README.md]:
```suggestion
However, the Datadog Operator is still in alpha, so it is not yet a recommended way of installing the Agent in production. Datadog's end goal is to support both the Helm chart and the operator (as well as providing basic deployment files in the Agent [repository][4] when CRDs and Helm are not an option) as official ways of installing Datadog.
```

File: README.md [hunk 10]
```diff
@@ -10,70 +10,42 @@
...
-- Create/Update/Delete ExtendedDaemonSet <Namespace/Name>
-- Create/Update/Delete Deployment <Namespace/Name>
-- Create/Update/Delete ClusterRole </Name>
-- Create/Update/Delete Role <Namespace/Name>
-- Create/Update/Delete ClusterRoleBinding </Name>
-- Create/Update/Delete RoleBinding <Namespace/Name>
-- Create/Update/Delete Secret <Namespace/Name>
-- Create/Update/Delete PDB <Namespace/Name>
-- Create/Delete ServiceAccount <Namespace/Name>
+- [Configure and provide Custom check to the Agents][6].
```
Comment (cswatt) [inline README.md]:
```suggestion
- [Configure and provide custom checks to the Agents][6].
```

File: README.md [hunk 11]
```diff
@@ -10,70 +10,42 @@
...
+- [Deploy the Datadog Cluster Agent with your node Agents][7]
```
Comment (cswatt) [inline README.md]:
```suggestion
- [Deploy the Datadog Cluster Agent with your node Agents][7].
```

File: docs/cluster_agent_setup.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
+# Cluster Agent
+
+To deploy the Cluster Agent along with the Agent update the current `DatadogAgent` resource with the [`datadog-agent-with-clusteragent.yaml` file][1]:
```
Comment (cswatt) [inline docs/cluster_agent_setup.md]:
```suggestion
To deploy the Cluster Agent along with the Agent, update the current `DatadogAgent` resource with the [`datadog-agent-with-clusteragent.yaml` file][1]:
```

File: docs/cluster_agent_setup.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+  clusterAgent:
+    image:
+      name: "datadog/cluster-agent:latest"
+    config:
+      metricsProviderEnabled: true
+      clusterChecksEnabled: true
+    replicas: 2
+```
+
+**Note**: `<DATADOG_CLUSTER_AGENT_TOKEN>` is a custom 32 characters long token that you can defined, if omited, a random one is generated automatically.
```
Comment (cswatt) [inline docs/cluster_agent_setup.md]:
```suggestion
**Note**: `<DATADOG_CLUSTER_AGENT_TOKEN>` is a custom 32 characters long token that you can define. If it is omitted, a random one is generated automatically.
```

File: docs/custom_check.md [hunk 14]
```diff
@@ -0,0 +1,154 @@
+# Custom Check
+
+To configure one of Datadog 400+ integrations, leverage the [Agent Autodiscovery feature][1]. But if you want to run a Custom Check the `DatadogAgent` resource can be configured to provide custom checks (`checks.d`) and their configuration files (`conf.d`) at initialization time. A `ConfigMap` resource needs to be configured for each of these settings before the `DatadogAgent` resource using them is created.
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
To configure one of Datadog's 400+ integrations, leverage the [Agent Autodiscovery feature][1]. But if you want to run a custom check, the `DatadogAgent` resource can be configured to provide custom checks (`checks.d`) and their configuration files (`conf.d`) at initialization time. A `ConfigMap` resource needs to be configured for each of these settings before the `DatadogAgent` resource using them is created.
```

File: docs/custom_check.md [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+
+Below is an example of configuring these `ConfigMaps` for a single check `hello` that submits the `hello.world` metrics to Datadog. See the [Introduction to Integrations][2] to learn what is a check in the Datadog ecosystem.
+
+## Create the check files
+
+At its core a check needs a configuration file `hello.yaml` and a script file `hello.py`:
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
A check needs a configuration file `hello.yaml` and a script file `hello.py`:
```

File: docs/custom_check.md [hunk 16]
```diff
@@ -0,0 +1,154 @@
...
+   class HelloCheck(AgentCheck):
+       def check(self, instance):
+           self.gauge('hello.world', 1, tags=['env:dev'])
+   ```
+
+## Create the check ConfigMaps
+
+Once the `hello` check files are created, create the associated `ConfigMaps`:
+
+1. Create the config map for the custom check yaml configuration file `hello.yaml`:
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
1. Create the ConfigMap for the custom check YAML configuration file `hello.yaml`:
```

File: docs/custom_check.md [hunk 17]
```diff
@@ -0,0 +1,154 @@
...
+
+   ```shell
+   $ kubectl create configmap -n $DD_NAMESPACE confd-config --from-file=hello.yaml
+   configmap/confd-config created
+   ```
+
+2. Verify that the config map has been correctly created:
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
2. Verify that the ConfigMap has been correctly created:
```

File: docs/custom_check.md [hunk 18]
```diff
@@ -0,0 +1,154 @@
...
+       init_config:
+
+       instances: [{}]
+   kind: ConfigMap
+   metadata:
+     name: confd-config
+     namespace: datadog
+   ```
+
+3. Create the config map for the custom check python file `hello.py`:
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
3. Create the ConfigMap for the custom check Python file `hello.py`:
```

File: docs/custom_check.md [hunk 19]
```diff
@@ -0,0 +1,154 @@
...
+
+   ```shell
+   $ kubectl create configmap -n $DD_NAMESPACE checksd-config --from-file=hello.py
+   configmap/checksd-config created
+   ```
+
+4. Verify that the config map has been correctly created:
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
4. Verify that the ConfigMap has been correctly created:
```

File: docs/custom_check.md [hunk 20]
```diff
@@ -0,0 +1,154 @@
...
+  agent:
+    image:
+      name: "datadog/agent:latest"
+    confd:
+      configMapName: "confd-config"
+    checksd:
+      configMapName: "checksd-config"
+```
+
+**Note**: ConfigMaps created needs to be in the same `DD_NAMESPACE` as the `DatadogAgent` resource.
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
**Note**: ConfigMaps need to be in the same `DD_NAMESPACE` as the `DatadogAgent` resource.
```

File: docs/custom_check.md [hunk 21]
```diff
@@ -0,0 +1,154 @@
...
+
+This deploys the Datadog Agent with your custom check.
+
+### Multiple checks
+
+In order to populate `ConfigMaps` with content of multiple checks or their respective configurations files, the following approach can be used:
```
Comment (cswatt) [inline docs/custom_check.md]:
```suggestion
In order to populate `ConfigMaps` with the content of multiple checks or their respective configurations files, the following approach can be used:
```

File: docs/data_collected.md [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+| `datadog.operator.agent.deployment.success`              | gauge       | `1` if the desired number of Agent replicas equals the number of available Agent pods, `0` otherwise.                               |
+| `datadog.operator.clusteragent.deployment.success`       | gauge       | `1` if the desired number of Cluster Agent replicas equals the number of available Cluster Agent pods, `0` otherwise.               |
+| `datadog.operator.clustercheckrunner.deployment.success` | gauge       | `1` if the desired number of Cluster Check Runner replicas equals the number of available Cluster Check Runner pods, `0` otherwise. |
+| `datadog.operator.reconcile.success`                     | gauge       | `1` if the last recorded reconcile error is null, `0` otherwise. The `reconcile_err` tag describes the last recorded error.         |
+
+**Note:** The [Datadog API and APP keys][1] are required to forward metrics to Datadog, they must be provided in the `credentials` field in the Custom Resource definiton.
```
Comment (cswatt) [inline docs/data_collected.md]:
```suggestion
**Note:** The [Datadog API and app keys][1] are required to forward metrics to Datadog. They must be provided in the `credentials` field in the Custom Resource definition.
```

File: docs/data_collected.md [hunk 23]
```diff
@@ -0,0 +1,38 @@
...
+
+The Datadog Operator exposes Golang and Controller metrics in OpenMetrics format. For now they can be collected using the [OpenMetrics integration][2]. A Datadog integration will be available in the future.
+
+The OpenMetrics check is activated by default via [autodiscovery annotations][3] and is scheduled by the Agent running on the same node as the Datadog Operator Pod.
```
Comment (cswatt) [inline docs/data_collected.md]:
```suggestion
The OpenMetrics check is activated by default via [Autodiscovery annotations][3] and is scheduled by the Agent running on the same node as the Datadog Operator Pod.
```

File: docs/getting_started.md [hunk 24]
```diff
@@ -0,0 +1,161 @@
...
+   DD_NAMEOP="ddoperator"
+   ```
+
+4. Create the namespace:
+
+   ```shell
+   kubectl create ns $DD_NAMESPACE
+   ```
+
+5. Install the operator with helm:
```
Comment (cswatt) [inline docs/getting_started.md]:
```suggestion
5. Install the operator with Helm:
```

**Pull request merged**